components/autofill/src/error.rs

```diff
@@ -44,4 +70,82 @@ pub enum Error {
     NoSuchRecord(String),
 }
 
-pub type Result<T> = std::result::Result<T, Error>;
+// Define how our internal errors are handled and converted to external errors.
+impl GetErrorHandling for Error {
+    type ExternalError = AutofillApiError;
+
+    // Return how to handle our internal errors
+    fn get_error_handling(&self) -> ErrorHandling<Self::ExternalError> {
+        // WARNING: The details inside the `AutofillApiError` we return should not
+        // contain any personally identifying information.
+        // However, because many of the string details come from the underlying
+        // internal error, we operate on a best-effort basis, since we can't be
+        // completely sure that our dependencies don't leak PII in their error
+        // strings.  For example, `rusqlite::Error` could include data from a
+        // user's database in their errors, which would then cause it to appear
+        // in our `AutofillApiError::SqlError` structs, log messages, etc.
+        // But because we've never seen that in practice we are comfortable
+        // forwarding that error message into ours without attempting to sanitize.
+        match self {
+            Self::OpenDatabaseError(e) => ErrorHandling::convert(AutofillApiError::SqlError {
+                reason: e.to_string(),
+            })
+            .report_error("autofill-open-database-error"),
```

Comment:
I erred on the side of making error reports for everything, just so that we can get some insight into how many errors we're getting.  After this is merged and in use on mobile, we should definitely check out the errors on sentry and consider turning some of this into `log_warning` or `log_info`.